ls1intum/Artemis
# `Assessment`: Add option to hide optional exercises in assessment dashboard

### Checklist
#### General


#### Client

### Motivation and Context


Artemis offers the option to not include exercises in the score of the course. These exercises might be used for example in tutor groups or for training purposes of students.
Some of these exercises are not graded at all, even though the students can participate and submit in these exercises. For example a modeling exercise that is discussed at the end of the tutor group.
Additionally tutors sometimes want to focus on the exercises that matter for the students' scores first since they might be more important than optional exercises.

### Description

This PR adds a filter to the assessment dashboard to filter all optional exercises.


### Steps for Testing

Prerequisites:
- 1 Instructor
- 3 exercises that need some kind of assessment (One with unfinished assessments, one fully assessed, one which is not included in the score of the course)
- An exam with at least one submission and two correction rounds (may be an old exam)

1. Log in to Artemis
2. Navigate to assessment dashboard
3. Make sure that both filters work
4. Check that you can still enable/disable the second assessment and sort the rows correctly

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





assessment-dashboard.component.ts: 93% -> 94%

### Screenshots
Completely unfiltered:
![Bildschirmfoto 2021-12-20 um 14 17 01](https://user-images.githubusercontent.com/38322605/146772964-685182cb-5713-4df9-9e5a-da0da2596622.png)
Finish exercises filtered out: (standard)
![Bildschirmfoto 2021-12-20 um 14 15 29](https://user-images.githubusercontent.com/38322605/146772837-aa5191b7-ba70-4d19-8e5e-80d7004f4643.png)
Finished and optional exercises filtered out:
![Bildschirmfoto 2021-12-20 um 14 15 34](https://user-images.githubusercontent.com/38322605/146772839-15d1fe7d-7cce-4764-a319-cdb6766d3a12.png)
